webapp/src/widgets/menu/subMenuOption.tsx

```diff
@@ -30,22 +34,44 @@ function SubMenuOption(props: SubMenuOptionProps): JSX.Element {
         }
     }, [isHovering])
 
+    const ref = useRef<HTMLDivElement>(null)
+
+    const styleRef = useRef<CSSProperties>({})
+
+    useEffect(() => {
+        const newStyle: CSSProperties = {}
+        if (props.position === 'auto' && ref.current) {
+            const openUp = MenuUtil.openUp(ref)
+            if (openUp.openUp) {
+                newStyle.bottom = 0
+            } else {
+                newStyle.top = 0
+            }
+        }
+
+        styleRef.current = newStyle
+    }, [ref.current])
+
     return (
         <div
             id={props.id}
-            className={`MenuOption SubMenuOption menu-option${openLeftClass}${isOpen ? ' menu-option-active' : ''}`}
+            className={`MenuOption SubMenuOption menu-option${openLeftClass}${isOpen ? ' menu-option-active' : ''} ${props.className || ''}`}
```

Comment:
This would prevent snapshot changes. 

```suggestion
            className={`MenuOption SubMenuOption menu-option${openLeftClass}${isOpen ? ' menu-option-active' : ''}${props.className ? props.className : ''}`}
```

Reply:
Done.